packages/quantic/force-app/main/default/lwc/quanticFeedbackModalQna/__tests__/quanticFeedbackModalQna.test.js

```diff
@@ -0,0 +1,299 @@
+    is: QuanticFeedbackModalQna,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+function lightningElementValueChange(element, selector, value) {
+  const lightningElement = element.shadowRoot.querySelector(selector);
+  // All base components in Jest tests are stubs. So we have to set the value manually.
+  // Including to the checkValidity method.
+  lightningElement.value = value;
+  lightningElement.validity = true;
+  lightningElement.checkValidity = () => true;
+  const event = new CustomEvent('change');
+  expect(lightningElement).not.toBeNull();
+  lightningElement.dispatchEvent(event);
+}
+
+describe('c-quantic-feedback-modal-qna', () => {
+  afterEach(() => {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    jest.clearAllMocks();
+  });
+
+  it('loads correctly', async () => {
+    const element = createTestComponent();
+    await flushPromises();
+
+    const answerEvaluationQuestions = element.shadowRoot.querySelectorAll(
+      selectors.answerEvaluationQuestions
+    );
+    expect(answerEvaluationQuestions.length).toBe(4);
+
+    const questionTopicRadio = element.shadowRoot.querySelector(
+      selectors.questionTopicRadio
+    );
+    expect(questionTopicRadio).not.toBeNull();
+
+    const questionHallucinationRadio = element.shadowRoot.querySelector(
+      selectors.questionHallucinationRadio
+    );
+    expect(questionHallucinationRadio).not.toBeNull();
+
+    const questionDocumentedRadio = element.shadowRoot.querySelector(
+      selectors.questionDocumentedRadio
+    );
+    expect(questionDocumentedRadio).not.toBeNull();
+
+    const questionReadableRadio = element.shadowRoot.querySelector(
+      selectors.questionReadableRadio
+    );
+    expect(questionReadableRadio).not.toBeNull();
+
+    const documentUrlInput = element.shadowRoot.querySelector(
+      selectors.documentUrlInput
+    );
+    expect(documentUrlInput).not.toBeNull();
+
+    const additionalDetailsInput = element.shadowRoot.querySelector(
+      selectors.additionalDetailsTextarea
+    );
+    expect(additionalDetailsInput).not.toBeNull();
+
+    const skipButton = element.shadowRoot.querySelector(selectors.skipButton);
+    expect(skipButton).not.toBeNull();
+
+    const submitButton = element.shadowRoot.querySelector(
+      selectors.submitButton
+    );
+    expect(submitButton).not.toBeNull();
+  });
+
+  describe('when submitting the feedback', () => {
+    describe('with missing required answers', () => {
+      it('should not submit the feedback and not show the success screen', async () => {
```

Comment:
As mentionned I would add a check for the required message here

Reply:
I can't. Jest lightning-* components don't actually "exist". They are only stubs in the DOM with some public attributes accessible to tests.

Meaning they don't actually react to display a required message while in unit tests.

Reply:
ok